lib/datasource/gitlab.js

```diff
@@ -0,0 +1,34 @@
+const glGot = require('../platform/gitlab/gl-got-wrapper');
+
+module.exports = {
+  getPreset,
+  getPkgReleases,
+};
+
+async function getPreset(pkgName, presetName = 'default') {
+  if (presetName !== 'default') {
+    throw new Error(
+      { pkgName, presetName },
+      'Sub-preset names are not supported with Gitlab datasource'
+    );
+  }
+  let res;
+  try {
+    const urlEncodedPkgName = encodeURIComponent(pkgName);
+    const url = `https://gitlab.com/api/v4/projects/${urlEncodedPkgName}/repository/files/renovate.json?ref=master`;
```

Comment:
As per @rtaum, he thinks GitLab makes us pick a branch. Ideally we could find a way to just tell GitLab "the default branch", in case it is not master. So let's see if we can figure out a way. Maybe it even takes one extra request to do?

Reply:
You can do this by doing an API call to `v4/projects/:id/repository/branches`:

https://docs.gitlab.com/ee/api/branches.html